### PR TITLE
feat(hypergraphs): add exact hypergraph edge-intersection profiles

### LINKED ISSUE #2261: [Operation]: Return exact hypergraph edge-intersection profiles and linearity witnesses
## Parent gap / RFC

Follow-up to #1742. That issue proposed exact edge-intersection profiles, but PR #2112 implemented the foundational hypergraph value, parameters, vertex degrees, dual, incidence graph, and clique expansion without this operation.

A circumradius-conflict investigation supplied a concrete use. Its six-uniform layer needed counts C_q of edge pairs meeting in exactly q vertices and needed to establish that a surviving hypergraph was linear after deleting one vertex from every non-linear pair. The asymptotic bounds remained proof reasoning, but finite instances still required a complete source-bound overlap ledger.

## Public mathematical postcondition

For one bounded indexed finite hypergraph, return the exact intersection of every unordered pair of distinct indexed edges, the complete intersection-size histogram, and the resulting linearity status with a violating edge pair when non-linear.

## Proposed operation ID

hypergraph.edge_intersections.compute

## Input

One existing FiniteHypergraph value. Distinct edge IDs remain distinct even when their member subsets are equal.

## Result

Return the source hypergraph together with:

- one canonical entry for every unordered pair of distinct edge IDs;
- the exact intersection vertex tuple and its cardinality for each pair;
- the total edge-pair count;
- the complete histogram mapping intersection size to pair count;
- the maximum pairwise intersection size;
- whether every two distinct indexed edges meet in at most one vertex; and
- when the hypergraph is not linear, the first canonical violating edge pair and its complete intersection.

The pair entries and histogram are authoritative. Maximum intersection and linearity are reconstructed derived values, not independently asserted scalars.

## Completeness semantics

Every accepted request returns an exact complete profile containing each unordered indexed edge pair exactly once. Empty and one-edge hypergraphs return an empty pair family, maximum intersection zero, and linear status.

The operation has no UNKNOWN or partial-result branch. Requests outside the operation-specific work or output envelope are rejected before pair materialization.

## Bounds and cost class

Retain the existing FiniteHypergraph limits of at most 100 vertices and 100 indexed edges, with an operation-specific preflight on:

- pair count m(m-1)/2;
- aggregate input incidences;
- worst-case intersection cells, bounded by the sum of min(|E_i|, |E_j|) over edge pairs;
- aggregate returned intersection memberships;
- label bytes; and
- serialized result bytes.

A conservative initial envelope can require the worst-case pair-intersection cell bound to be at most 65,536. Under the current 100-edge, 100-vertex value bounds, the direct kernel performs at most 4,950 bounded set intersections. Narrower admission limits should be explicit rather than discovered through memory pressure or transport overflow.

## Maintained implementation

Use a direct native kernel over immutable Python sets or bitsets. Iterate canonical edge-ID pairs, compute exact intersections, accumulate the histogram, and derive maximum intersection and linearity from the same entries. No external solver or backend is needed.

## Why this is not caller-side composition

The current hypergraph operations do not retain pairwise edge intersections. The incidence graph and clique expansion expose neighboring projections but lose the complete edge-pair overlap profile. Reconstructing it requires callers to enumerate every edge pair, perform repeated finite-set intersections, preserve edge-ID provenance, prove that no pair was omitted, and independently keep histogram and linearity claims synchronized.

This is one finite structural profile analogous to the existing complete degree profile. It is useful for linear hypergraphs, block designs, conflict systems, intersection graphs, and finite checks of extremal constructions without prescribing a proof strategy.

## Deferred operation family

This issue does not add:

- asymptotic overlap bounds or theorem application;
- random sampling or alteration;
- hypergraph independent-set, matching, cover, or transversal optimization;
- codegree profiles;
- line-graph construction;
- restriction, deletion, contraction, shadow, or blocker transforms;
- circumradius-specific conflict construction; or
- mutable search or hypergraph sessions.

Those require separate admission evidence or remain caller-side reasoning and composition.

### PR DESCRIPTION
**Pull request by morluto**
## Problem

The finite-hypergraph operations expose parameters, degrees, duals, incidence graphs, and clique expansions, but none retains the exact overlap of every pair of indexed edges. Reconstructing that profile caller-side means enumerating the complete pair family, preserving edge-ID provenance, and keeping a histogram, maximum, linearity decision, and counterexample synchronized independently.

Closes #2261.

## Solution

Add `hypergraph.edge_intersections.compute`, an exact complete profile over one canonical `FiniteHypergraph`.

The operation enumerates each declared edge pair `(i, j)` with `i < j`, so distinct edge IDs remain distinct even when their member subsets are equal. Each ledger entry contains the exact lexicographically ordered intersection and its size. The histogram, maximum intersection size, linearity flag, and first size-greater-than-one violation are reconstructed from that authoritative ledger. Empty and one-edge hypergraphs therefore have an empty histogram, maximum zero, and linear status without a special mathematical outcome.

The result retains its source hypergraph and validation recomputes the complete ledger. Source, pair, histogram, scalar, and witness mutations cannot produce a valid stale profile. Admission happens before pair materialization and charges the full pair family, aggregate incidences, exact returned intersection memberships, exact label bytes, and a conservative complete-result size.

The intersection-cell preflight is result-sensitive without doing pair intersections: a vertex of degree `d` occurs in exactly `C(d, 2)` edge-pair intersections. Summing that identity over vertices gives the exact complete-ledger cell count and the exact repeated-label byte contribution. This admits sparse-overlap families according to their actual result size while retaining a bounded preflight. The size calculation uses the same strict JSON encoder as dispatch, including non-normalized Unicode strings.

The kernel uses immutable Python sets. NetworkX 3.6.1 provides ordinary graph containers and graph-set intersection, not a finite hypergraph value or a source-bound edge-overlap profile. Adapting through it would add conversion without replacing the bounded set-intersection kernel, so this change adds no dependency or solver.

Suggested review order:

1. Work/output preflight and source-bound result model in `_models.py`.
2. Complete ledger kernel and native function in `_operations.py`.
3. Tool declaration, owner-local admission, and native namespace exports.
4. Defining-invariant, mutation, boundary, scale, and catalog tests.

## Testing

- `make check` — Ruff and mypy passed; 2,672 tests passed.
- `make test-math TESTS=tests/math/hypergraphs` — 56 passed.
- `uv run --locked pytest -n 0 tests/catalog/test_admission.py tests/catalog/test_catalog.py -q` — 18 passed.
- `uv run --locked pytest -n 0 tests/integration/catalog/test_builtin_examples.py tests/integration/catalog/test_public_operation_contract_conformance.py -k hypergraph.edge_intersections.compute -q` — 2 passed, 965 deselected.
- `uv run --locked pytest -n 0 tests/tooling/test_pytest_collection_ownership.py tests/math/hypergraphs/test_edge_intersections.py -q` — 25 passed on the original operation tree; the final owner lane above includes the added scale regression.

- Specialist validation run: public catalog example execution, accepted-request mutation conformance, owner-local catalog admission, and test-owner boundary validation, as listed above.

Coverage includes complete pair reconstruction, equal member sets under distinct edge IDs, disjoint pairs, empty and one-edge cases, linear and nonlinear profiles, source/result mutation rejection, a 4,950-pair accepted boundary with 64,350 returned memberships, the immediately larger rejected family, a 100-edge cyclic sparse-overlap family with exactly 9,100 returned memberships, escaped-label output overflow, and exact non-normalized Unicode byte accounting.

## Public contract impact

Adds `hypergraph.edge_intersections.compute`, the native `edge_intersections(FiniteHypergraph)` function, and public `EdgeIntersectionEntry` and `EdgeIntersectionsResult` values. It also exposes the existing `hypergraphs` domain through `jacobian.math`. Every accepted request returns one complete exact profile; out-of-envelope requests are rejected before execution rather than returning a partial or unknown result.

## Public operation admission

- Concrete gap: no operation returned the complete indexed edge-pair overlap relation or an exact linearity witness.
- Why existing operations or typed values are insufficient: the incidence graph and clique expansion are projections that discard the complete edge-pair overlap ledger, while parameters and degree profiles contain only scalar or vertex-local information.
- Stable mathematical result: every unordered indexed edge pair and its exact intersection, with a complete size histogram and derived maximum, linearity decision, and first canonical violation.
- Semantic domain and admitted execution envelope: the existing canonical finite hypergraph domain with at most 100 vertices and 100 indexed edges, unique labels and edge IDs of at most 64 code points, and declared edge memberships; at most 4,950 pairs, 10,000 source incidences, 65,536 exact returned intersection memberships, 51,200 aggregate UTF-8 label bytes, and a complete strict-JSON result estimate within 10 MiB.
- Controlling work, intermediate, memory, and output quantities: pair count bounds exhaustive pair execution and replay; `sum_v C(deg(v), 2)` is the exact aggregate returned-membership count; immutable edge sets use linear incidence storage; the authored result separately caps aggregate intersection memberships before nested parsing; repeated edge IDs, intersection labels, histogram data, and the optional witness are included in the preflight output bound.
- Exact representation, algorithm regimes, and maintained backend: direct `frozenset` intersections over the domain-owned immutable hypergraph value, in declared edge-pair order, followed by exact deterministic aggregation. There is one exhaustive regime and no external backend expression or approximation.
- Remaining fixed caps and their classification: the 100-vertex, 100-edge, and 64-code-point limits are existing representation caps; 65,536 exact intersection cells is a conservative operation work/output safety cap; and 10 MiB is the shared transport limit.
- Admission decision: `KEEP` — the complete source-bound overlap relation is a reusable finite structural primitive with materially stronger reliability than caller-authored pair enumeration.

## Closure matrix

None. Issue #2261 requests this single admitted operation; codegree profiles, line-graph construction, optimization, and hypergraph transforms remain outside its scope.

## Checklist

- [x] `make check` passes
- [x] Explicitly relevant specialist validation is listed above
- [x] Harbor task or verifier changes are not applicable
- [x] Public operation changes include an owner-local admission decision
- [x] Every accepted result is complete and exact; approximate, incomplete, unknown, and unavailable outcomes are not applicable
- [x] New bounds include defining boundary and realistic sparse-overlap scale evidence; there is one algorithm regime and no crossover
- [x] No shared abstraction was introduced

### CODE REVIEW

File: src/jacobian/math/hypergraphs/_models.py [hunk 1]
```diff
@@ -54,6 +70,250 @@ def require_valid_hypergraph(self) -> Self:
...
+
+    edge_count = len(hypergraph.edges)
+    pair_count = edge_count * (edge_count - 1) // 2
+    edge_sizes = tuple(len(members) for _, members in hypergraph.edges)
+    total_incidences = sum(edge_sizes)
+    intersection_cells = sum(
+        min(edge_sizes[left], edge_sizes[right])
+        for left in range(edge_count)
+        for right in range(left + 1, edge_count)
+    )
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/hypergraphs/_models.py]: Addressed in 074fabc53. Preflight now derives the exact aggregate memberships as sum over vertices of C(degree, 2) and uses the same multiplicities for strict-JSON byte accounting. The added 100-edge cyclic 14-uniform regression has exactly 9,100 cells and is now admitted; boundary and overflow tests plus make check pass.

**Comment by morluto** [comment]
@codex
